Source/JavaScriptCore/runtime/WaiterListManager.cpp

```diff
@@ -224,11 +222,6 @@ void WaiterListManager::notifyWaiterImpl(const AbstractLocker& listLocker, Ref<W
         });
         return;
     }
-
-    // If waiter is a SyncWaiter, we null out its vm to indicate that this waiter
-    // is removed from the WaiterList.
-    ASSERT(waiter->vm());
-    waiter->clearVM(listLocker);
```

Comment:
what is the reason of this change?

Reply:
Previously, we used whether `waiter->vm() == nullptr` as the indication of whether the waiter node is dequeued from the list or not. However, we can just use `waiter->isOnList()` since `waiter` has type `BasicRawSentinelNode`. See

```
bool didGetDequeued = !syncWaiter->isOnList();
```

in the above changes.

Reply:
So, `syncWaiter->vm()` check in the while loop is not correct, right?

Reply:
I think given that this patch's intention is not changing that part, using the current mechanism is better.